scrypto/src/resource/vault.rs

```diff
@@ -14,36 +14,44 @@ use crate::engine::{api::*, call_engine, types::VaultId};
 use crate::math::*;
 use crate::misc::*;
 use crate::resource::*;
-use crate::sfunctions;
+use crate::sfunctions2;
 use crate::types::*;
 
 #[derive(Debug, TypeId, Encode, Decode)]
-pub enum VaultMethod {
-    Put(Bucket),
-    Take(Decimal),
-    TakeNonFungibles(BTreeSet<NonFungibleId>),
-    CreateProof(),
-    CreateProofByAmount(Decimal),
-    CreateProofByIds(BTreeSet<NonFungibleId>),
-    GetAmount(),
-    GetResourceAddress(),
-    GetNonFungibleIds(),
+pub struct VaultPutInput {
+    pub bucket: Bucket,
 }
 
-impl VaultMethod {
-    pub fn name(&self) -> &str {
-        match self {
-            VaultMethod::Put(_) => "put",
-            VaultMethod::Take(_) => "take",
-            VaultMethod::TakeNonFungibles(_) => "take_non_fungibles",
-            VaultMethod::CreateProof() => "create_proof",
-            VaultMethod::CreateProofByAmount(_) => "create_proof_by_amount",
-            VaultMethod::CreateProofByIds(_) => "create_proof_by_ids",
-            VaultMethod::GetAmount() => "get_amount",
-            VaultMethod::GetResourceAddress() => "get_resource_address",
-            VaultMethod::GetNonFungibleIds() => "get_non_fungible_ids",
-        }
-    }
+#[derive(Debug, TypeId, Encode, Decode)]
+pub struct VaultTakeInput {
+    pub amount: Decimal,
+}
+
+#[derive(Debug, TypeId, Encode, Decode)]
+pub struct VaultTakeNonFungiblesInput {
+    pub non_fungible_ids: BTreeSet<NonFungibleId>,
+}
+
+#[derive(Debug, TypeId, Encode, Decode)]
+pub struct VaultGetAmountInput {}
+
+#[derive(Debug, TypeId, Encode, Decode)]
+pub struct VaultGetResourceAddressInput {}
+
+#[derive(Debug, TypeId, Encode, Decode)]
+pub struct VaultGetNonFungibleIdsInput {}
+
+#[derive(Debug, TypeId, Encode, Decode)]
+pub struct VaultCreateProofInput {}
+
+#[derive(Debug, TypeId, Encode, Decode)]
+pub struct VaultCreateProofByAmountInput {
+    pub amount: Decimal,
+}
+
+#[derive(Debug, TypeId, Encode, Decode)]
+pub struct VaultCreateProofByIdsInput {
+    pub ids: BTreeSet<NonFungibleId>,
 }
 
```

Comment:
I was wondering if it would be better to add some namespace for these input/output types, like encapsulate with a `mod` or `impl`. Not a big fan of prefix. :)